### PR TITLE
[Type hint] PNDM schedulers

### PR DESCRIPTION
**Pull request by daspartho**
Based on 24, 25, and 26 of issue https://github.com/huggingface/diffusers/issues/287

@anton-l could you check it? 
Thanks :)

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_pndm.py [hunk 1]
```diff
@@ -51,12 +51,12 @@ class PNDMScheduler(SchedulerMixin, ConfigMixin):
...
-        num_train_timesteps=1000,
-        beta_start=0.0001,
-        beta_end=0.02,
-        beta_schedule="linear",
-        tensor_format="pt",
-        skip_prk_steps=False,
+        num_train_timesteps: int = 1000,
+        beta_start: float = 0.0001,
+        beta_end: float = 0.02,
+        beta_schedule: Optional[str] = "linear",
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_pndm.py:57]:
```suggestion
        beta_schedule: str = "linear",
```

File: src/diffusers/schedulers/scheduling_pndm.py [hunk 2]
```diff
@@ -51,12 +51,12 @@ class PNDMScheduler(SchedulerMixin, ConfigMixin):
...
+        tensor_format: Optional[str] = "pt",
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_pndm.py:58]:
```suggestion
        tensor_format: str = "pt",
```

File: src/diffusers/schedulers/scheduling_pndm.py [hunk 3]
```diff
@@ -97,7 +97,7 @@ def __init__(
         self.tensor_format = tensor_format
         self.set_format(tensor_format=tensor_format)
 
-    def set_timesteps(self, num_inference_steps, offset=0):
+    def set_timesteps(self, num_inference_steps: int, offset=0):
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_pndm.py:100]:
```suggestion
    def set_timesteps(self, num_inference_steps: int, offset: int = 0):
```

File: src/diffusers/schedulers/scheduling_pndm.py [hunk 4]
```diff
@@ -264,7 +264,12 @@ def _get_prev_sample(self, sample, timestep, timestep_prev, model_output):
 
         return prev_sample
 
-    def add_noise(self, original_samples, noise, timesteps):
+    def add_noise(
+        self,
+        original_samples: Union[torch.FloatTensor, np.ndarray],
+        noise: Union[torch.FloatTensor, np.ndarray],
+        timesteps: Union[torch.IntTensor, np.ndarray],
+    ):
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_pndm.py:272]:
```suggestion
    ) -> torch.Tensor:
```

**Review by patrickvonplaten — COMMENTED** [review]
Thanks a lot for the PR! Looks good - left a couple of suggestions :-)

File: src/diffusers/schedulers/scheduling_pndm.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/diffusers/schedulers/scheduling_pndm.py:58]: Think the `tensor_format` cannot be `None`

File: src/diffusers/schedulers/scheduling_pndm.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/diffusers/schedulers/scheduling_pndm.py:57]: `beta_schedule` has to be string I think :-)

**Comment by daspartho** [comment]
@patrickvonplaten made the necessary changes :)

**Review by patrickvonplaten — APPROVED** [review]
Very cool! Thanks

**Pull request merged**